### PR TITLE
Enhance gmt convert's ability to detect duplicate records

### PR DESCRIPTION
**Pull request by PaulWessel**
**Description of proposed changes**

gmt **convert** already has the option **-T**[**h**][**d**] for eliminating **h**eaders on output and/or to eliminate exact **d**uplicate records.  However, what is a duplicate to the user may not be an exact replicate of a record but perhaps consecutive records that have the same values in one or more columns.  For instance, a time-series with identical times may be problematic in a processing workflow so those should be excluded, and perhaps duplicate (lon,lat) values should be eliminated as representing the same point.

This PR extends this option to  **-T**[**h**][**d**[_select_]], where the optional _select_ argument is the usual list of columns that can be comma-separated or ranges, or regular stepping, all possibly inverted (see **-Q** for similar syntax).  One can also add the column _t_ to include the trailing text in this comparison.  If not given we default to matching all the columns (as we currently do).  Below are some examples of use with a test data set:

```
0	0	4.46031835138	0.275015041826	AAA
1	1	4.5705316237	2.1778079668	BBB
1	1	1.92499958767	2.41849793467	BBB
1	1	3.46806994568	2.69478794313	CCC
4	4	3.46806994568	2.69478794313	DDD
5	5	1.45537732935	0.237260430696	DDD
5	5	0.526774435084	2.63605870895	AAA
5	7	4.71711448613	4.23872130655	EEE
8	8	3.46806994568	0.188999134204	AAA
9	9	3.46806994568	2.69478794313	AAA
```

##  Eliminate duplicated x-values

```
gmt convert data.txt -Td0
0	0	4.46031835138	0.275015041826	AAA
1	1	4.5705316237	2.1778079668	BBB
4	4	3.46806994568	2.69478794313	DDD
5	5	1.45537732935	0.237260430696	DDD
8	8	3.46806994568	0.188999134204	AAA
9	9	3.46806994568	2.69478794313	AAA
```

##  Eliminate duplicated x/y-values

```
gmt convert data.txt -Td0:1
0	0	4.46031835138	0.275015041826	AAA
1	1	4.5705316237	2.1778079668	BBB
4	4	3.46806994568	2.69478794313	DDD
5	5	1.45537732935	0.237260430696	DDD
5	7	4.71711448613	4.23872130655	EEE
8	8	3.46806994568	0.188999134204	AAA
9	9	3.46806994568	2.69478794313	AAA
```

##  Eliminate repeated z-values

```
gmt convert data.txt -Td2
0	0	4.46031835138	0.275015041826	AAA
1	1	4.5705316237	2.1778079668	BBB
1	1	1.92499958767	2.41849793467	BBB
1	1	3.46806994568	2.69478794313	CCC
5	5	1.45537732935	0.237260430696	DDD
5	5	0.526774435084	2.63605870895	AAA
5	7	4.71711448613	4.23872130655	EEE
8	8	3.46806994568	0.188999134204	AAA
```

##  Eliminate repeated values in cols 2 and 3:

```
gmt convert data.txt -Td2,3
0	0	4.46031835138	0.275015041826	AAA
1	1	4.5705316237	2.1778079668	BBB
1	1	1.92499958767	2.41849793467	BBB
1	1	3.46806994568	2.69478794313	CCC
5	5	1.45537732935	0.237260430696	DDD
5	5	0.526774435084	2.63605870895	AAA
5	7	4.71711448613	4.23872130655	EEE
8	8	3.46806994568	0.188999134204	AAA
9	9	3.46806994568	2.69478794313	AAA

```

## Eliminate duplicate record solely on the basis of trailing text

```
gmt convert data.txt -Tdt
0	0	4.46031835138	0.275015041826	AAA
1	1	4.5705316237	2.1778079668	BBB
1	1	3.46806994568	2.69478794313	CCC
4	4	3.46806994568	2.69478794313	DDD
5	5	0.526774435084	2.63605870895	AAA
5	7	4.71711448613	4.23872130655	EEE
8	8	3.46806994568	0.188999134204	AAA
```

## Eliminate duplicate record defined by col 2 and trailing text

```
gmt convert data.txt -Td2,t
0	0	4.46031835138	0.275015041826	AAA
1	1	4.5705316237	2.1778079668	BBB
1	1	1.92499958767	2.41849793467	BBB
1	1	3.46806994568	2.69478794313	CCC
4	4	3.46806994568	2.69478794313	DDD
5	5	1.45537732935	0.237260430696	DDD
5	5	0.526774435084	2.63605870895	AAA
5	7	4.71711448613	4.23872130655	EEE
8	8	3.46806994568	0.188999134204	AAA
```

### CODE REVIEW

**Comment by seisman** [comment]
Add some tests for the new feature?

**Comment by PaulWessel** [comment]
I can probably write a text test that compares the output to these saved results.  Should not have to add a plot test for a module that is purely data.  Tomorrow.

**Comment by PaulWessel** [comment]
> Add some tests for the new feature?

Added duplicates.sh with the 6 example tests listed above.

File: src/gmtconvert.c [hunk 1]
```diff
@@ -93,6 +96,8 @@ struct GMTCONVERT_CTRL {
 	} S;
 	struct GMTCONVERT_T {	/* -T[sd] */
```
Comment (seisman) [inline src/gmtconvert.c:97]:
```suggestion
	struct GMTCONVERT_T {	/* -T[h][d[[~]selection]] */
```

**Pull request merged**